front-end-notes/grunt.md

```diff
@@ -0,0 +1,38 @@
+# Grunt
+
+### What is Grunt?
+
+GruntJS is a JavaScript based command line build tool that helps developers automate repetitive tasks. It performs tasks like minification, compilation, unit testing, linting, etc. As more and more development moves towards the client side, it makes a lot of sense to use tools that help developers be more productive.
+
+### GruntJS Prerequisites
+
+GruntJS requires NodeJS version 0.8.0 or above. You will need to use the npm package manager to install GruntJS and its plugins. Newer versions of NodeJS have the NPM package manager as part of the default distribution. Download the latest version of NodeJS from the website https://nodejs.org/en/.
+
+### Install Grunt
+
+On your command line type: ``npm install -g grunt-cli``
+
+#### Create a Gruntfile.js in your project directory
+
+Next, open Gruntfile.js and put in the following code:
+
+If your .js files are in a directory other than javascripts be sure to replace the javascripts in grunt.initConfig with the name of your directory.
+```
+module.exports = function(grunt) {
+
+ grunt.initConfig({
+   jshint: {
+     files: ['./javascripts/**/*.js']
+   },
+ });
+
+ require('matchdep').filterDev('grunt-*').forEach(grunt.loadNpmTasks);
+ grunt.registerTask('default', ['jshint', 'watch']);

```

Comment:
Grunt will not run as is, as you have a task _watch_ in the _grunt.registerTask_ but no task named _watch_ in the task list, only _jshint_